bentoml/_internal/frameworks/lightgbm.py

```diff
@@ -176,121 +166,96 @@ def save(
             params, lgb_train, num_boost_round=20, valid_sets=lgb_eval
         )
 
+        # `save` the booster to BentoML modelstore:
         tag = bentoml.lightgbm.save("my_lightgbm_model", gbm, booster_params=params)
-    """  # noqa
-    context: t.Dict[str, t.Any] = {
-        "framework_name": "lightgbm",
-        "pip_dependencies": [f"lightgbm=={get_pkg_version('lightgbm')}"],
-    }
+    """  # noqa: LN001
+
+    # Ensure that `model` is actually the Booster object, and not for example one of the scikit-learn wrapper objects.
+    if not isinstance(model, lgb.basic.Booster):
+        # Deal with a LightGBM issue (https://github.com/microsoft/LightGBM/issues/3014)
+        if not hasattr(model, "fitted_"):
+            model.fitted_ = True
```

Comment:
Is this necessary if we're extracting the booster from the sklearn wrapper?

Reply:
So I have a LGBMClassifier model (sklearn-wrapper) that was trained and saved with LightGBM 3.0.0. When I unpickle it and try to directly access the booster, I get an exception:
```
In [2]: clf.booster_
---------------------------------------------------------------------------
NotFittedError                            Traceback (most recent call last)
Input In [2], in <cell line: 1>()
----> 1 clf.booster_

File ~/.virtualenvs/bentoml-lightgbm/lib/python3.8/site-packages/lightgbm/sklearn.py:854, in LGBMModel.booster_(self)
    852 """Booster: The underlying Booster of this model."""
    853 if not self.__sklearn_is_fitted__():
--> 854     raise LGBMNotFittedError('No booster found. Need to call fit beforehand.')
    855 return self._Booster

NotFittedError: No booster found. Need to call fit beforehand.
```

But the model is fitted of course. According to that issue linked to in the comment, this is due to some misconfiguration in the attributes, which was fixed in a release post 3.0.0.

Reply:
Then should we set the support for lightgbm to be >=3.0.0? I feel like we don't want to coupling lightgbm logics here.

Reply:
It would have to be >3.0.0 (not >=), because the issue is still present in 3.0.0. But that's fine, I can understand not wanting to couple to lightgbm internals. I guess we'll just have to ask people (like myself) that have a <=3.0.0 model to deal with this issue themselves. Not a big deal.

Reply:
I think it's fine to set fitted ourselves, I didn't realize `booster_` would actually check that it's fitted.